cpp/arcticdb/version/test/test_version_map_batch.cpp

```diff
@@ -0,0 +1,433 @@
+#include <gtest/gtest.h>
+
+#include <arcticdb/version/version_map_batch_methods.hpp>
+#include <arcticdb/stream/test/stream_test_common.hpp>
+
+using namespace arcticdb;
+using namespace arcticdb::pipelines;
+
+struct VersionMapBatchStore : arcticdb::TestStore {
+protected:
+    std::string get_name() override {
+        return "version_map_batch";
+    }
+};
+
+namespace arcticdb {
+
+AtomKey test_index_key(StreamId id, VersionId version_id) {
+    return atom_key_builder().version_id(version_id).creation_ts(PilotedClock::nanos_since_epoch()).content_hash(3)
+        .start_index(4).end_index(5).build(id, KeyType::TABLE_INDEX);
+}
+
+void add_versions_for_stream(
+    std::shared_ptr<VersionMap> version_map,
+    std::shared_ptr<Store> store,
+    StreamId stream_id,
+    size_t num_versions,
+    size_t start = 0u) {
+    for(auto i = start; i < start + num_versions; ++i) {
+        version_map->write_version(store, test_index_key(stream_id, i));
+    }
+};
+}
+
+TEST_F(VersionMapBatchStore, SimpleVersionIdQueries) {
```

Comment:
Comment from William, from the other PR: `I mind repetition much less in tests than I do in production code. Personally I feel like some of the rules like DRY and no explicit strings that are mandatory for real code are actually reversed in test code, where you want everything to be obvious and local; I don't really want to have to go and look elsewhere to see what a test is doing when it fails.`

Reply:
I think most of your suggestions are fair. Only for the first one, be aware that this change is neither exposing any extra API nor using this functionality in any existing one. Let me have a second thought about trying to have the test with less duplication, which is what I agree most.

Reply:
The reason why it was decided to stress the tests with multiple requests (not just 3), is because we are running it in parallel and for the (sometimes fragile) way folly works, we wanted to be sure that there is no any hidden bug there when we do a massive parallelisation.